test/python/circuit/test_controlled_gate.py

```diff
@@ -1577,7 +1581,7 @@ def test_single_controlled_rotation_gates(self, gate, cgate):
         elif gate.name == "rz":
             self.assertLessEqual(uqc.size(), 43, f"\n{uqc}")
         else:
-            self.assertLessEqual(uqc.size(), 20, f"\n{uqc}")
+            self.assertLessEqual(uqc.size(), 23, f"\n{uqc}")
```

Comment:
Why does the size increase in this test? shouldn't the decomposition be more efficient? (I admit I haven't looked in depth so I might be missing something)

Reply:
should we use size? perhaps it's better to count cx-count and u-count desperately, and check that cx-count does not increase

Reply:
btw, is there some randomization in this test that can explain this?

Reply:
The 3 additional gates were `U(0,0,0)` gates, it's not very clear to me why that would've changed, most likely a different decomposition path.. but @ShellyGarion's suggestion sounds good, what we care about are the CX gates, I'll update the test to check that.

Reply:
done in ffbb955c3